R/utilities.R

```diff
@@ -4517,23 +4474,78 @@ classify_dlbcl_chapuy <- function(
 
-    # If no SV data is provided, get the SVs from GAMBL
-    if(missing(sv_data)){
-        message("No SV data is provided.")
-        message("Will retreive SVs available through GAMBL.")
 
-        sv_data <- get_combined_sv(
-            sample_ids = these_samples_metadata$sample_id,
-            oncogenes = chapuy_features$sv_features,
-            projection = projection
-        )
+    # Now use this data to classify the samples according to one of the systems
+    if(method=="chapuy"){
+
+      # SVs are needed only for Chapuy predictions
+      # If no SV data is provided, get the SVs from GAMBL
+      if(missing(sv_data)){
+          message("No SV data is provided.")
+          message("Will retreive SVs available through GAMBL.")
+
+          sv_data <- get_combined_sv(
+              sample_ids = these_samples_metadata$sample_id,
+              oncogenes = chapuy_features$sv_features,
+              projection = projection
+          )
+      }
+
+      sv_data <- sv_data %>%
+          annotate_sv(
+              genome_build = projection
+          ) %>%
+          dplyr::filter(!is.na(partner))
+
+      predictions <- classify_dlbcl_chapuy(
+          these_samples_metadata,
+          maf_data,
+          seg_data,
+          sv_data,
+          projection = projection,
+          output = output
+      )
+
+    }else if (method=="lacy") {
+      predictions <- classify_dlbcl_lacy(
+          these_samples_metadata,
+          maf_data,
+          seg_data,
+          projection = projection,
+          output = output,
+          include_N1 = include_N1
+      )
+    }else{
+      stop("You requested an unvalid method. Please choole one of chapuy or lacy")
     }
-    sv_data <- sv_data %>%
-        annotate_sv(
-            genome_build = projection
-        ) %>%
-        dplyr::filter(!is.na(partner))
 
+    return(predictions)
+
+}
+
+
+
+#' Classify DLBCLs according to genetic subgroups of Chapuy et al.
+#'
+#' Use the feature weights from NMF model to assemble the binary matrix and classify DLBCL tumors based on C0-C5 system of Chapuy et al
+#'
+#' @param these_samples_metadata The metadata data frame that contains sample_id column with ids for the samples to be classified.
+#' @param maf_data The MAF data frame to be used for matrix assembling. At least must contain the first 45 columns of standard MAF format.
+#' @param seg_data The SEG data frame to be used for matrix assembling. Must be of standard SEG formatting, for example, as returned by get_sample_cn_segments.
+#' @param sv_data The SV data frame to be used for matrix assembling. Must be of standard BEDPE formatting, for example, as returned by get_combined_sv.
+#' @param projection The projection of the samples. Only used to retrerive data through GAMBLR when it is not provided. Defaults to grch37.
+#' @param output The output to be returned after prediction is done. Can be one of predictoins, matrix, or both. Defaults to predictions.
```

Comment:
Instead of "both" as an option what about just returning both by default if the user doesn't specify "output" explicitly?

Reply:
I have changed this to both so the default output now is matrix and the predictions